chiselFrontend/src/main/scala/chisel3/core/Bits.scala

```diff
@@ -67,6 +90,7 @@ sealed abstract class Bits(dirArg: Direction, width: Width, override val litArg:
     if (isLit()) {
       Bool(((litValue() >> x.toInt) & 1) == 1)
     } else {
+      Binding.checkSynthesizable(this, s"'this' ($this)")
```

Comment:
The 'worst case' scenario I outlined is that you accidentally generate a valid FIRRTL AST and FIRRTL doesn't error. Even a line-by-line FIRRTL check wouldn't catch this because (by 'luck') the same reference was valid in both Modules.

Even in the easier cases, it is convenient to know definitely whether you have an undefined reference or improper cross-module reference.

Also, if the FIRRTL semantics change, wouldn't you have to update Chisel 3 anyway to emit the proper new constructs?